fe/fe-core/src/main/java/com/starrocks/qe/DnsCache.java

```diff
@@ -0,0 +1,48 @@
+// Copyright 2021-present StarRocks, Inc. All rights reserved.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     https://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package com.starrocks.qe;
+
+import com.google.common.collect.Maps;
+import com.starrocks.thrift.TNetworkAddress;
+import org.apache.commons.validator.routines.InetAddressValidator;
+
+import java.net.InetAddress;
+import java.net.UnknownHostException;
+import java.util.Map;
+
+public class DnsCache {
+    private final Map<String, String> cache = Maps.newHashMap();
+
+    public String lookup(String host) throws UnknownHostException {
+        if (InetAddressValidator.getInstance().isValidInet4Address(host)) {
+            return host;
+        }
+        String ip = cache.get(host);
+        if (ip != null) {
+            return ip;
+        }
+        ip = InetAddress.getByName(host).getHostAddress();
```

Comment:
You don't need a map here. Java, by default, has 30-second DNS cache TTL.  By the way, for those who want to explicitly check the default TTL, here is the [Code](https://github.com/AdoptOpenJDK/openjdk-jdk11/blob/19fb8f93c59dfd791f62d41f332db9e306bc1422/src/java.base/share/classes/sun/net/InetAddressCachePolicy.java#L48) where you can check the default TTL time.

Here is a nice article about java DNS cache:
https://medium.com/@avocadi/dns-cache-ttl-in-java-adb1c0525459

Reply:
FE StarRocksMain, change it to 60 seconds. so when a pod is rebuilt with a different ip address, it may take 60s in the worst case to get the right ip address from dns server.